cartoframes/datasets.py

```diff
@@ -0,0 +1,252 @@
+            raise CartoException('Cannot create table: {}.'.format(job['failed_reason']))
+
+    def _cartodbfy_query(self):
+        return "SELECT CDB_CartodbfyTable('{org}', '{table_name}')" \
+            .format(org=(self.cc.creds.username() if self.cc.is_org else 'public'),
+                    table_name=self.table_name)
+
+    def _copyfrom(self, with_lonlat=None):
+        util_cols = Dataset.UTIL_COLS
+        geom_col = get_geom_col_name(self.df)
+
+        columns = ','.join(norm_colname(c) for c in self.df.columns if c not in util_cols)
+        self.cc.copy_client.copyfrom(
+            """COPY {table_name}({columns},the_geom)
+               FROM stdin WITH (FORMAT csv, DELIMITER '|');""".format(table_name=self.table_name, columns=columns),
+            self._rows(self.df, self.df.columns, with_lonlat, geom_col)
+        )
+
+    def _rows(self, df, cols, with_lonlat, geom_col):
+        geom_cols = Dataset.SUPPORTED_GEOM_COL_NAMES
+        for i, row in df.iterrows():
+            csv_row = ''
+            the_geom_val = None
+            for col in cols:
+                if with_lonlat and col in geom_cols:
+                    continue
+                val = row[col]
+                if pd.isnull(val) or val is None:
+                    val = ''
+                if with_lonlat:
+                    if col == with_lonlat[0]:
+                        lng_val = row[col]
+                    if col == with_lonlat[1]:
+                        lat_val = row[col]
+                if col == geom_col:
+                    the_geom_val = row[col]
+                else:
+                    csv_row += '{val}|'.format(val=val)
+
+            if the_geom_val is not None:
+                geom = decode_geom(the_geom_val)
+                if geom:
+                    csv_row += 'SRID=4326;{geom}'.format(geom=geom.wkt)
+            if with_lonlat is not None:
+                csv_row += 'SRID=4326;POINT({lng} {lat})'.format(lng=lng_val, lat=lat_val)
+
+            csv_row += '\n'
+            yield csv_row.encode()
+
+    def _drop_table_query(self):
+        return '''DROP TABLE IF EXISTS {table_name}'''.format(table_name=self.table_name)
+
+    def _create_table_query(self, with_lonlat=None):
+        util_cols = Dataset.UTIL_COLS
+        if with_lonlat is None:
+            geom_type = get_geom_col_type(self.df)
+        else:
+            geom_type = 'Point'
+
+        col = ('{col} {ctype}')
+        cols = ', '.join(col.format(col=norm_colname(c),
+                                    ctype=dtypes2pg(t))
+                         for c, t in zip(self.df.columns, self.df.dtypes) if c not in util_cols)
+
+        if geom_type:
+            cols += ', {geom_colname} geometry({geom_type}, 4326)'.format(geom_colname='the_geom', geom_type=geom_type)
+
+        create_query = '''CREATE TABLE {table_name} ({cols})'''.format(table_name=self.table_name, cols=cols)
+        return create_query
+
+
+def norm_colname(colname):
```

Comment:
Some failing cases that cartodbfy is not going to like very much: `á`, `public`, `_a`.

We can always improve this later though.

Reply:
I created an issue

Reply:
Other issue on the same topic: https://github.com/CartoDB/cartoframes/issues/427